utils/http.go

```diff
@@ -112,6 +176,20 @@ func buildAccessRecords(repo string, actions ...string) []auth.Access {
 	return requiredAccess
 }
 
+func buildCatalogRecord(actions ...string) []auth.Access {
+	requiredAccess := make([]auth.Access, 0, len(actions))
+	for _, action := range actions {
+		requiredAccess = append(requiredAccess, auth.Access{
+			Resource: auth.Resource{
+				Type: "registry",
+				Name: "catalog",
```

Comment:
After more discussion with @jlhawn, this resource should always be requested with the "*" permission, so I've hard coded that into this function. The configured "pull" action will now only be used when requesting access to the changefeed of a specific repo.